File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Tasks/stopandresumetask.rb
module Asposetasksjava
  module StopAndResumeTask
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Create a ChildTasksCollector instance
		collector = Rjb::import('com.aspose.tasks.ChildTasksCollector').new

		# Collect all the tasks from RootTask using TaskUtils
		Rjb::import('com.aspose.tasks.TaskUtils').apply(project.getRootTask(), collector, 0)

		tasks = collector.getTasks()

		tsk = Rjb::import('com.aspose.tasks.Tsk')

		# Parse through all the collected tasks
		i = 0
		while i < tasks.size()
			task = tasks.get(i)
		    if task.get(tsk.STOP).toString() == "1/1/2015"
				puts "NA"
			else
				puts "Task Stop: " + task.get(tsk.STOP).toString()
			end

			if task.get(tsk.RESUME).toString() == "1/1/2015"
				puts "NA"
			else
				puts "Task Resume: " + task.get(tsk.RESUME).toString()
			end	  
		    puts "---------------------------------------------"
		    i +=1
		end
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/StopAndResumeTask.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\ChildTasksCollector as ChildTasksCollector;
use com\aspose\tasks\TaskUtils as TaskUtils;
use com\aspose\tasks\Tsk as Tsk;
class StopAndResumeTask{
    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Create a ChildTasksCollector instance
		$collector = new ChildTasksCollector();

            # Collect all the tasks from RootTask using TaskUtils
        $taskUtils=new TaskUtils();
        $taskUtils->apply($project->getRootTask(), $collector, 0);

		$tasks = $collector->getTasks();

		$tsk = new Tsk();

		# Parse through all the collected tasks
		$i = 0;
		while ($i < sizeof($tasks)) {
            $task = $tasks->get($i);
            if ((string)$task->get($tsk->STOP) == "1/1/2015") {
                print "NA" . PHP_EOL;
            } else {
                print "Task Stop: " . (string)$task->get($tsk->STOP) . PHP_EOL;// . toString()
            }

            if ((string)$task->get($tsk->RESUME) == "1/1/2015") {
                print "NA" . PHP_EOL;
            } else {
                print "Task Resume: " . (string)$task->get($tsk->RESUME);// . toString()
            }
            print "---------------------------------------------" . PHP_EOL;
            $i += 1;
        }
    }
}
?>